src/index.jsx

```diff
@@ -49,27 +55,77 @@ function Form() {
         setPref('app', app);
     }, [app])
 
+    const menuItems = Object.entries(AVAILABLE_APPS).map(([name, value]) => ({
+        text: value.title,
+        key: name
+    }));
+    
+    const handleMenuChange = (item) => {
+        if (item) {
+            setApp(item.key);
+        }
+    };
+
+    const changeOpenState = (item) => {
+        if(isDropdownOpen == true){
+            document.getElementById("root").className = "normal";
+            document.getElementById("popoutBody").className = "normal";
+            setDropdownOpen(false);
+        } else {
+            document.getElementById("root").className = "expanded";
+            document.getElementById("popoutBody").className = "expanded";
+            setDropdownOpen(true);
+        }
+        
+    };
+
     return <Box display="flex" flexDirection="column">
+        {/* https://datahub.berkeley.edu */}
         <Heading sx={{ fontSize: 2, mb: 1 }}>JupyterHub URL</Heading>
 
-        <TextInput value={hubUrl} onChange={(ev) => setHubUrl(ev.target.value)} placeholder="https://myjupyterhub.org" aria-label="JupyterHub URL" />
+        <TextInput
+            value={hubUrl}
+            onChange = {
+                (ev) => setHubUrl(ev.target.value)
+            }
+            placeholder="https://myjupyterhub.org"
+            aria-label="JupyterHub URL"
+            sx={{ pt: 0.5, pb: 0.5 }}
+        />
+
         <Text color="danger.fg" sx={{ visibility: isValidHubUrl ? "hidden" : "visible" }}>Enter a valid URL</Text>
 
-        <Heading sx={{ fontSize: 2, mb: 1, mt: 2 }}>Open in</Heading>
-        <select className="form-select mb-1" onChange={(ev) => setApp(ev.target.value)} value={app}>
-            {Object.entries(AVAILABLE_APPS).map(([name, value]) => {
-                return <option key={name} value={name}>{value.title}</option>
-            })};
-        </select>
-
-        <Button disabled={!isValidHubUrl || finishedCopying} sx={{ mt: 2 }} onClick={() => {
-            copyGeneratedUrl(hubUrl, app);
-            // Flash a 'Copied!' message for 3 seconds after copying
-            setFinishedCopying(true);
-            setTimeout(() => setFinishedCopying(false), 3 * 1000)
-        }}>
-            <CopyIcon /> {finishedCopying ? "Copied!" : "Copy nbgitpuller link"}
-        </Button>
+        <Heading sx={{ fontSize: 2, mb: 1, mt: 3 }}>Open in</Heading>
+
+        <DropdownMenu
```

Comment:
Ah, I can see how this `<DropdownMenu>` component is actually requiring all the manual height toggling, otherwise it gets clipped.

<img width="424" alt="image" src="https://github.com/yuvipanda/nbgitpuller-link-generator-webextension/assets/30430/a2840383-4de5-4645-83e8-48fe85505dab">

This is actually not a problem with the regular HTML `<select>` control:

<img width="380" alt="image" src="https://github.com/yuvipanda/nbgitpuller-link-generator-webextension/assets/30430/00031af9-ca88-4cd3-825e-1704cba148e8">

I think having to manually specify pixel based height values long term causes maintenance issues, and I think the extra prettiness of the dropdown component isn't worth it. So I'd suggest moving this back to the HTML `<select>` element.

Reply:
reverted to <select>